modules/util/portal-tools-sample-sql-builder/src/main/java/com/liferay/portal/tools/sample/sql/builder/DataFactory.java

```diff
@@ -1190,108 +1199,251 @@ public AssetEntryModel newAssetEntryModel(
 			journalArticleLocalizationModel.getTitle());
 	}
 
-	public List<PortletPreferencesModel>
-		newAssetPublisherPortletPreferencesModels(long plid) {
+	public AssetListEntryModel newAssetListEntryModel(long groupId, int index) {
+		AssetListEntryModel assetListEntryModel = new AssetListEntryModelImpl();
 
-		return ListUtil.fromArray(
-			newPortletPreferencesModel(plid, BlogsPortletKeys.BLOGS),
-			newPortletPreferencesModel(plid, JournalPortletKeys.JOURNAL));
+		// PK fields
+
+		assetListEntryModel.setAssetListEntryId(_counter.get());
+
+		// Group instance
+
+		assetListEntryModel.setGroupId(groupId);
+
+		// Audit fields
+
+		assetListEntryModel.setCompanyId(_companyId);
+		assetListEntryModel.setUserId(_sampleUserId);
+		assetListEntryModel.setUserName(_SAMPLE_USER_NAME);
+		assetListEntryModel.setCreateDate(new Date());
+		assetListEntryModel.setModifiedDate(new Date());
+
+		// Other fields
+
+		String title = StringBundler.concat(
+			"dynamic-collection-", groupId, "-", index);
+
+		Class<?> clazz = AssetEntry.class;
```

Comment:
@slnn This seems only used once, please inline.

Reply:
OK

Reply:
Update to the second commitLPD-45665 Generate AssetListEntry for each AssetPublisher portlet